vere/sist.c

```diff
@@ -1359,6 +1359,22 @@ u3_sist_boot(void)
 
       if ( 0 != u3_Host.ops_u.key_c ) {
         u3_noun des = u3m_file(u3_Host.ops_u.key_c);
+
+        // handle trailing newline
+        //
+        {
+          c3_c* key = u3r_string(des);
+          c3_w len = strlen(key);
+
+          if (len && (key[len - 1] == '\n')) {
+            key[len - 1] = '\0';
+            //////////////////////////////// "free" des first?
+            des = u3i_string(key);
```

Comment:
As you've guessed, this leaks references as written. The `u3r_string()` call above does not take ownership of the reference to `des` (ie, it uses *retain* semantics, not *transfer*), so we have to lose our reference to `des` before re-assigning it. Simply `u3z(des);` before the re-assignment.